pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -0,0 +1,438 @@
+			members = append(members, memberResult(cluster, memberStatus))
+			continue
+		}
+
+		// take only the overall conditions from the memberstatus (ie. don't include status of member operator, host connection, etc.) in order to keep readability.
+		// member status details can be viewed on the member cluster directly.
+		memberStatus := toolchainv1alpha1.MemberStatusStatus{Conditions: memberStatusObj.Status.Conditions}
+		for _, condition := range memberStatusObj.Status.Conditions {
+			if condition.Type == toolchainv1alpha1.ConditionReady && condition.Status != "True" {
+				// the existing memberstatus condition is added to the toolchainstatus but the condition is not ready so set the component error to bubble up the error to the overall toolchain status
+				componentError = errs.Wrap(fmt.Errorf(condition.Message), fmt.Sprintf("member cluster %s not ready", cluster.Name))
+			}
+		}
+		members = append(members, memberResult(cluster, memberStatus))
+	}
+
+	// add member cluster statuses to toolchainstatus
+	toolchainStatus.Status.Members = members
+	return componentError
+}
+
+// updateStatusConditions updates Member status conditions with the new conditions
+func (r *ReconcileToolchainStatus) updateStatusConditions(memberStatus *toolchainv1alpha1.ToolchainStatus, newConditions ...toolchainv1alpha1.Condition) error {
+	// the controller should always update at least the last updated timestamp of the status so the status should be updated regardless of whether
+	// any specific fields were updated. This way a problem with the controller can be indicated if the last updated timestamp was not updated.
+	conditionsWithTimestamps := []toolchainv1alpha1.Condition{}
+	for _, condition := range newConditions {
+		condition.LastTransitionTime = metav1.Now()
+		condition.LastUpdatedTime = &metav1.Time{Time: condition.LastTransitionTime.Time}
+		conditionsWithTimestamps = append(conditionsWithTimestamps, condition)
+	}
+	memberStatus.Status.Conditions = conditionsWithTimestamps
+	return r.client.Status().Update(context.TODO(), memberStatus)
+}
+
+func (r *ReconcileToolchainStatus) setStatusReady(toolchainStatus *toolchainv1alpha1.ToolchainStatus) error {
+	return r.updateStatusConditions(
+		toolchainStatus,
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.ConditionReady,
+			Status: corev1.ConditionTrue,
+			Reason: toolchainv1alpha1.ToolchainStatusAllComponentsReadyReason,
+		})
+}
+
+func (r *ReconcileToolchainStatus) setStatusNotReady(toolchainStatus *toolchainv1alpha1.ToolchainStatus, message string) error {
+	return r.updateStatusConditions(
+		toolchainStatus,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.ConditionReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  toolchainv1alpha1.ToolchainStatusComponentsNotReadyReason,
+			Message: message,
+		})
+}
+
+func customMemberStatus(conditions ...toolchainv1alpha1.Condition) toolchainv1alpha1.MemberStatusStatus {
```

Comment:
since you always pass a single `condition` in this func, you could rename it to:
```
func memberStatus(condition toolchainv1alpha1.Condition) 
```
and do something like:
```
members = append(members, memberResult(cluster, memberStatus(*memberStatusNotFoundCondition)))
``` 
above
(but that's just a suggestion to make code more compact)

Reply:
While it does make it more compact, conditions is still an array in the API and I figured we may want to add other conditions in the future.

Reply:
ok, sounds good to me ;)